src/Sulu/Bundle/MediaBundle/DependencyInjection/ImageFormatCompilerPass.php

```diff
@@ -74,13 +78,18 @@ protected function setFormatsFromFile($fullPath, &$activeFormats, $defaultOption
         $fileName = basename($fullPath);
 
         $locator = new FileLocator($folder);
-        $loader = new XmlFormatLoader($locator);
-        $loader->setDefaultOptions($defaultOptions);
+
+        $xmlLoader10 = new XmlFormatLoader10($locator);
+        $xmlLoader11 = new XmlFormatLoader11($locator);
+        $xmlLoader10->setGlobalOptions($defaultOptions);
+        $xmlLoader11->setGlobalOptions($defaultOptions);
+
+        $resolver = new LoaderResolver([$xmlLoader10, $xmlLoader11]);
+        $loader = new DelegatingLoader($resolver);
```

Comment:
When looking at this again (sry, haven't seen it before...) this should be located in the constructor instead of this method. This method is called every single time, and it creates new objects every time, which is not necessary.